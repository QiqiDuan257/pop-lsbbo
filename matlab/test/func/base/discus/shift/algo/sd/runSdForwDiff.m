clear; clc;

% nohup matlab -nodesktop -nodisplay -nosplash <runSdForwDiff.m >runSdForwDiff.out 2>&1 &

%% load the `minFunc` library, which is downloaded from
%   https://www.cs.ubc.ca/~schmidtm/Software/minFunc.html
%   https://www.cs.ubc.ca/~schmidtm/Software/copyright.html
%   https://www.cs.ubc.ca/~schmidtm/Software/minFunc_2012.zip
% first put runSdForwDiff.m in the working directory of the `minFunc` library
addpath(genpath(pwd));

%%
startRun = tic;
randSeed = 20190819; % for reproducibility
funcName = @popSFuncDiscus_;
funcDims = [2 10 100 1000 2000 4000 6000 8000 10000]';
% funcDims = [2 10]';
optRes = cell(length(funcDims), 1);
for d = 1 : length(funcDims)
    startRunD = tic;
    funcDim = funcDims(d);
    maxFEval = 1e5 * funcDim;
    rng(randSeed + d);
    x0 = -100 + 100 * rand(funcDim, 1);
    options = [];
    options.Method = 'sd';
    options.numDiff = 1; % forward-differencing
    options.MaxFunEvals = maxFEval;
    options.MaxIter = maxFEval;
    options.Display = 'off';
    [x, y, exitFlag, output] = minFunc(funcName, x0, options);
    res = {};
    res.opty = y;
    res.optx = x;
    res.exitFlag = exitFlag;
    res.output = output;
    optRes{d} = res;
    fprintf('%6d: opty %7.2e + runtime %7.2e.\n', ...
        funcDim, y, toc(startRunD));
end
save('runSdForwDiff.mat', 'optRes', 'funcDims', '-v7.3');
fprintf('Total runtime: %7.5e.\n', toc(startRun));
