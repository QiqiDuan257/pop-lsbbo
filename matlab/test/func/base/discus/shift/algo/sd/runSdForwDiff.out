
                            < M A T L A B (R) >
                  Copyright 1984-2016 The MathWorks, Inc.
                   R2016b (9.1.0.441655) 64-bit (glnxa64)
                             September 7, 2016

#############################################################################
#             $$$ Enjoy the <pop-lsbbo> pure-MATLAB Library $$$             #
#                        --- version: 0.00.000.1 ---                        #
#############################################################################
     2: opty 2.64e+00 + runtime 2.23e+00.
    10: opty 2.73e+01 + runtime 2.60e+00.
   100: opty 5.54e+02 + runtime 1.44e+01.
  1000: opty 2.22e+03 + runtime 2.23e+02.
  2000: opty 4.25e+03 + runtime 9.87e+02.
  4000: opty 1.80e+04 + runtime 1.90e+03.
  6000: opty 2.71e+04 + runtime 5.89e+03.
  8000: opty 2.42e+04 + runtime 2.62e+03.
 10000: opty 3.89e+04 + runtime 1.47e+04.
Total runtime: 2.63574e+04.
